### PR TITLE
RSE-326: Fix: Error responses including stacktraces

### PR DESCRIPTION
**Pull request by L2JE**
Ticket: https://pagerduty.atlassian.net/browse/RSE-326

#### Problem
When there's an internal error, the responses include information about the internal system (stacktraces).
image
or
image

#### Solution
Use an Interceptor that matches responses with internal server errors and render a default error page for any of them:
image

##### Previous behavior can be re-enabled for debug:
```
# In rundeck-config.properties to disable the interceptor
rundeck.feature.debug.showTracesOnResponse=false
```

### CODE REVIEW

**Comment by ronaveva** [comment]
1. Is it possible to use the ConfigurationService? (instead of grailsApplication.config)
2. Is it possible to log the exception? sometimes this information is actually useful to find the root cause of an issue.

**Comment by L2JE** [comment]
> 1. Is it possible to use the ConfigurationService? (instead of grailsApplication.config)
> 
> 2. Is it possible to log the exception? sometimes this information is actually useful to find the root cause of an issue.

1. I thought it could impact throughput but now I will run a stress test with and without using the ConfigurationService to see if it really does before making a change.

2. Even the unhandled exceptions are logged as usual before the `Interceptor.after` method is reached. The logging will depend on the rest of the infrastructure.

**Comment by L2JE** [comment]
There's no notable difference between using a controller in the interceptor to access properties and not using it:

#### 10 concurrent users accessing `/user/login` 10000 times with the default behavior (`rundeck.feature.debug.showTracesOnResponse=false`)
image

**RC1**: where properties are accessed using `grailsApplication.config.getProperty('rundeck.feature.debug.showTracesOnResponse', 'false')`

**RC2**: where properties are accessed using `this.configurationService.getBoolean('feature', 'debug', 'showTracesOnResponse', false)`

**Review by ronaveva — APPROVED** [review]
LGTM, just make sure to add the new property to the doc.

File: rundeckapp/src/test/groovy/rundeck/interceptors/InternalErrorsInterceptorSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        defineBeans {
+            configurationService(ConfigurationService) {
+                grailsApplication = grailsApplication
+            }
+        }
+
+        withRequest(controller:"demo", action: 'index')
+
+        then:
+        !interceptor.doesMatch()
```
Comment (ronaveva) [inline rundeckapp/src/test/groovy/rundeck/interceptors/InternalErrorsInterceptorSpec.groovy:48]: shouldn't the response have some sort of exception?

**Review by ronaveva — CHANGES_REQUESTED** [review]
please check the 2 minor comments

**Review by ronaveva — APPROVED** [review]
LGTM

**Pull request merged**